vowpalwabbit/core/src/reductions/cb/cb_explore_adf_large_action_space.cc

```diff
@@ -318,6 +320,76 @@ void cb_explore_adf_large_action_space::randomized_SVD(const multi_ex& examples)
   }
 }
 
+std::pair<float, uint64_t> cb_explore_adf_large_action_space::find_max_volume(uint64_t X_rid, Eigen::MatrixXf& X)
+{
+  // Finds the max volume by replacing row X[X_rid] with some row in U.
+  // Returns the max volume, and the row id of U used for replacing X[X_rid].
+
+  float max_volume = 0.0f;
+  uint64_t U_rid = -1;
+  Eigen::RowVectorXf original_row = X.row(X_rid);
+
+  for (uint64_t i = 0; i < U.rows(); ++i)
+  {
+    X.row(X_rid) = U.row(i);
+    float volume = abs(X.determinant());
+    if (volume > max_volume) max_volume = volume, U_rid = i;
+    X.row(X_rid) = original_row;
+  }
+
+  assert(U_rid != -1);
+  return {max_volume, U_rid};
+}
+
+std::vector<bool> cb_explore_adf_large_action_space::compute_spanner()
+{
+  // Implements the C-approximate barycentric spanner algorithm in Figure 2 of the following paper
+  // Awerbuch & Kleinberg STOC'04: https://www.cs.cornell.edu/~rdk/papers/OLSP.pdf
+
+  assert(U.cols() == _d);
+  Eigen::MatrixXf X = Eigen::MatrixXf::Identity(_d, _d);
+  std::vector<uint64_t> action_indices(_d);
```

Comment:
looks like `action_indices` and `spanner_bitmap` can be member variables since their size depends on `_d` and cleaned up and re-utilized every call

Reply:
done